pkg/resource/record_set/hooks.go

```diff
@@ -0,0 +1,257 @@
+package record_set
```

Comment:
Since `ChangeResourceRecordSets` takes in a ChangeBatch (i.e. batches of record sets), this file mostly is comprised of of helper functions that are used to shape our CR specs into a ChangeBatch object.